azurelinuxagent/common/protocol/imds.py

```diff
@@ -296,9 +334,9 @@ def get_compute(self):
         """
 
         # ensure we get a 200
-        success, resp = self.get_metadata('instance/compute', is_health=False)
+        success, service_error, resp = self.get_metadata('instance/compute', is_health=False)
         if not success:
-            raise HttpError(resp)
+            raise ValueError(resp)
```

Comment:
Why is it a value error here?

Reply:
I think it was a naïve way to simplify logic. I'll revert.

Reply:
If you feel this needs a special error, then I would recommend creating an IMDSError class, which you can use to simplify and reuse later.